### PR TITLE
Consul cache

### PR DESCRIPTION
**Pull request by aureamunoz**
I pushed to my fork because I'm not able to do it to yours (permission denied)

### CODE REVIEW

File: service-discovery/consul/src/test/java/io/smallrye/stork/servicediscovery/consul/ConsulServiceDiscoveryIT.java [hunk 1]
```diff
@@ -64,12 +104,29 @@ void shouldGetServiceFromConsul() throws InterruptedException {
         assertThat(instances.get()).hasSize(1);
         assertThat(instances.get().get(0).getHost()).isEqualTo("example.com");
         assertThat(instances.get().get(0).getPort()).isEqualTo(8406);
+
+        deregisterService(serviceName);
+
+        // When the refresh interval is reached
+        Thread.sleep(65000);
```
Comment (michalszynkiewicz) [inline service-discovery/consul/src/test/java/io/smallrye/stork/servicediscovery/consul/ConsulServiceDiscoveryIT.java:111]: can you set the expiration in config to 1s or 5s so taht we don't have to wait that long in tests?

Comment (aureamunoz) [reply service-discovery/consul/src/test/java/io/smallrye/stork/servicediscovery/consul/ConsulServiceDiscoveryIT.java]: Done, I had to change the time unit to seconds also

Comment (michalszynkiewicz) [reply service-discovery/consul/src/test/java/io/smallrye/stork/servicediscovery/consul/ConsulServiceDiscoveryIT.java]: I think setting the interval to `5s` should work, without appending `s` before parsing

Comment (aureamunoz) [reply service-discovery/consul/src/test/java/io/smallrye/stork/servicediscovery/consul/ConsulServiceDiscoveryIT.java]: done! can you review Michal?

File: service-discovery/consul/src/test/java/io/smallrye/stork/servicediscovery/consul/ConsulServiceDiscoveryIT.java [hunk 2]
```diff
@@ -36,24 +36,69 @@
...
+
     }
 
     @Test
-    void shouldGetServiceFromConsul() throws InterruptedException {
+    void shouldRefetchWhenRefreshPeriodReached() throws InterruptedException {
+        //Given a service `my-service` registered in consul and a refresh-period of 5 seconds
         String serviceName = "my-service";
+        TestConfigProvider.addServiceConfig("my-service", null, "consul",
+                null, Map.of("consul-host", "localhost", "consul-port", String.valueOf(consulPort), "refresh-period", "5"));
```
Comment (michalszynkiewicz) [inline service-discovery/consul/src/test/java/io/smallrye/stork/servicediscovery/consul/ConsulServiceDiscoveryIT.java:93]: I thought we need `5s` or sth like this here. Is it seconds by default?

Comment (michalszynkiewicz) [reply service-discovery/consul/src/test/java/io/smallrye/stork/servicediscovery/consul/ConsulServiceDiscoveryIT.java:93]: or millis?

Comment (michalszynkiewicz) [reply service-discovery/consul/src/test/java/io/smallrye/stork/servicediscovery/consul/ConsulServiceDiscoveryIT.java:93]: oh, nevermind, I see 
```
        if (DIGITS.asPredicate().test(refreshPeriod)) {
            return Duration.ofSeconds(Long.valueOf(refreshPeriod));
        }
```

**Review by michalszynkiewicz — APPROVED** [review]

**Pull request merged**